runtime_scan/pkg/provider/docker/client.go

```diff
@@ -0,0 +1,353 @@
+	}
+
+	// Get container assets
+	containerAssets, err := c.getContainers(ctx)
+	if err != nil {
+		return nil, provider.FatalError{
+			Err: fmt.Errorf("failed to get containers. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	// Combine assets
+	assets := append(imageAssets, containerAssets...)
+
+	return assets, nil
+}
+
+func (c *Client) RunAssetScan(ctx context.Context, config *provider.ScanJobConfig) error {
+
+	err := c.prepareScanVolume(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to prepare scan volume. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	err = c.createScanConfigFile(config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create scanconfig.yaml file. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	containerId, err := c.createScanContainer(ctx, config)
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to create scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	err = c.dockerClient.ContainerStart(ctx, containerId, types.ContainerStartOptions{})
+	if err != nil {
+		return provider.FatalError{
+			Err: fmt.Errorf("failed to start scan container. Provider=%s: %w", models.Docker, err),
+		}
+	}
+
+	return nil
+}
+
+func (c *Client) prepareScanVolume(ctx context.Context, config *provider.ScanJobConfig) error {
+	scanName, err := getScanName(config)
+	if err != nil {
+		return err
+	}
+
+	// Create volume if not found
+	resp, err := c.dockerClient.VolumeList(ctx, volume.ListOptions{
+		Filters: filters.NewArgs(filters.Arg("name", scanName)),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get volumes. Provider=%s: %w", models.Docker, err)
+	}
+	if len(resp.Volumes) == 1 {
+		fmt.Printf("scan volume already created. Provider=%s", models.Docker)
+		return nil
+	}
+	if len(resp.Volumes) == 0 {
+		_, err = c.dockerClient.VolumeCreate(ctx, volume.CreateOptions{
+			Name: scanName,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to create volume. Provider=%s: %w", models.Docker, err)
+		}
+	}
+
+	readCloser, err := c.export(ctx, config)
+	if err != nil {
+		return fmt.Errorf("failed to export target. Provider=%s: %w", models.Docker, err)
+	}
+
+	// Create an ephemeral container to populate volume with export output
+	_, err = c.dockerClient.ImagePull(ctx, "alpine", types.ImagePullOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to pull helper image. Provider=%s: %w", models.Docker, err)
+	}
+	response, err := c.dockerClient.ContainerCreate(ctx,
+		&container.Config{
+			Image: "alpine",
+		},
+		&container.HostConfig{
+			Mounts: []mount.Mount{
+				{
+					Type:   mount.TypeVolume,
+					Source: scanName,
+					Target: "/data",
+				},
+			},
+		}, nil, nil, "")
+	if err != nil {
+		return fmt.Errorf("failed to create helper container. Provider=%s: %w", models.Docker, err)
+	}
+	defer func() {
+		err = c.dockerClient.ContainerRemove(ctx, response.ID, types.ContainerRemoveOptions{})
```

Comment:
If the container gets stuck in an infinite loop, I am not sure that this will remove it. You can specify to forcefully remove it via remove options type.